### PR TITLE
Remove hard limit on LIMIT

### LINKED ISSUE #1125: 1 Million limit on lettuceRediSearchClient.findall
this works at 1000000 but fails at 10000001
 Set<ProductEntity> products = lettuceRediSearchClient.findAll(1000001).resultStream()

com.rnbwarden.redisearch.LettuceTest > testFindAll FAILED
    io.lettuce.core.RedisCommandExecutionException at LettuceTest.java:184
        Caused by: io.lettuce.core.RedisCommandExecutionException at ExceptionFactory.java:135

this is the code link:
https://github.com/rnbWarden/jredisearch-spring-boot-starter/blob/master/src/test/java/com/rnbwarden/redisearch/LettuceTest.java

### PR DESCRIPTION
**Pull request by ashtul**
Was added in 1.6
fix #1125

### CODE REVIEW

**Comment by MeirShpilraien** [comment]
@mnunberg why this limit was added from the first place?

**Comment by mnunberg** [comment]
I am guessing so that someone didn't request a ton of items from the server and cause it to buffer the entire database in memory. I am not sure though that this is a good approach, since it would depend on the system; and also there are other ways for aggregation to weigh down the server.

**Comment by MeirShpilraien** [comment]
@mnunberg if its a breaking change compare to 1.4 then we need to revert it.

**Comment by MeirShpilraien** [comment]
@ashtul @mnunberg we need to remove the 1M limit only when used with aggregate

**Comment by ashtul** [comment]
@MeirShpilraien @mnunberg 
Added flag to enable with aggregate

File: src/pytest/test_aggregate.py [hunk 1]
```diff
@@ -390,6 +390,11 @@ def testReducerGeneratedAliasing(self):
                           'SORTBY', 2, '@brand', 'ASC')
         self.env.assertEqual('__generated_aliasfirst_valuetitle,by,price,desc', rv[1][2])
 
+    def testIssue1125(self):
```
Comment (MeirShpilraien) [inline src/pytest/test_aggregate.py]: Let's also have a test for ft.search that shows that the limit is working there ..

**Pull request merged**